### DISCUSSION TITLE
How to get dependencies of served pages?

### DISCUSSION THREAD

**Discussion by pana-exfo** [squidfunk/mkdocs-material#6870] [Q&A]
Hi

When serving pages, depending of your configuration there are dependencies to .js and .ts files.

What is most convenient way to get these dependencies listed? purl would be perfect.

Regards,
Antti

**Comment by kamilkrzyskow** [comment]
Hello @pana-exfo,
I'm not sure what you mean. The theme's source is written in TypeScript, but it's bundled into one bundle.js file and put into the Python package, before you install it locally. Something like the [`navigation.instant`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#instant-loading) feature can be enabled globally, but the logic to handle it is always inside the bundle, not enabling it won't prevent the [dependency](https://github.com/squidfunk/mkdocs-material/tree/master/src/templates/assets/javascripts/integrations/instant) from being served, as it was bundled long before that.

If that's not what you meant, then please elaborate.

**Reply by pana-exfo** [reply]
Hi @kamilkrzyskow,

This is related to identifying vulnerabilities in 3rd party code. For example some plugins use javascript libraries.

Is there a way to get a list of all dependencies?
- In minimum: name, version
- Ideal: [purl](https://github.com/package-url/purl-spec) of each package
